utilities/xmlrpcpp/src/XmlRpcValue.cpp

```diff
@@ -610,16 +611,52 @@ namespace XmlRpc {
       default:           break;
       case TypeBoolean:  os << _value.asBool; break;
       case TypeInt:      os << _value.asInt; break;
-      case TypeDouble:   os << _value.asDouble; break;
+      case TypeDouble:
+        {
+          static std::once_flag once;
+          int ret, required_size;
+          char buf[128]; // Should be long enough
+          ret = std::snprintf(buf, sizeof(buf)-1,
+                  getDoubleFormat().c_str(), _value.asDouble);
+          if (ret < 0) {
+            std::call_once(once,
+              [](){XmlRpcUtil::error("Failed to format with %s, using default %%.16g format",
+                getDoubleFormat().c_str());});
+            required_size = std::snprintf(buf, sizeof(buf)-1, "%.16g", _value.asDouble);
+          } else {
+            required_size = ret;
+          }
+          if (required_size >= 0 && required_size < (int) sizeof(buf)) {
+            buf[sizeof(buf)-1] = 0;
+            os << buf;
+          } else if (required_size >= (int) sizeof(buf)) {
+            int ret;
+            char required_buf[required_size+1];
+            ret = std::snprintf(required_buf, required_size,
+                    getDoubleFormat().c_str(), _value.asDouble);
+            if (ret == required_size) {
+              required_buf[required_size] = 0;
+              os << required_buf;
+            }
+          } else {
+            XmlRpcUtil::error("Unexpected error to format %s", getDoubleFormat().c_str());
+          }
+          break;
+        }
       case TypeString:   os << *_value.asString; break;
       case TypeDateTime:
         {
           struct tm* t = _value.asTime;
+          int ret;
           char buf[20];
-          std::snprintf(buf, sizeof(buf)-1, "%4d%02d%02dT%02d:%02d:%02d", 
-            t->tm_year,t->tm_mon,t->tm_mday,t->tm_hour,t->tm_min,t->tm_sec);
-          buf[sizeof(buf)-1] = 0;
-          os << buf;
+          ret = std::snprintf(buf, sizeof(buf)-1, "%4d%02d%02dT%02d:%02d:%02d",
+                  t->tm_year,t->tm_mon,t->tm_mday,t->tm_hour,t->tm_min,t->tm_sec);
+          if (ret > 0) {
+            buf[sizeof(buf)-1] = 0;
+            os << buf;
+          } else {
+            XmlRpcUtil::error("Unexpected error to format TypeDateTime");
+          }
```

Comment:
Why should the date time logic we changed at all? Can you describe a case where the `snprintf()` would fail? If not, please leave this part of the code unchanged.